### PR TITLE
Add support for automatically mapping used framework for generated up…

### PR DESCRIPTION
**Pull request by Azurelol**
## Description

Adds support for adding the framework used on a package onto the generated package directory name.

## Changes
* ![ADD] `PaketUnityInstall` now uses the framework used when generating the upm package directory
* ![IMPROVE] Major refactors on `PaketUnityInstall` to improve readability and several corner cases where it did not behave as expected across platforms
* ![REMOVE] Unused `assemblyDefinitionFileStrategy`

[NEW]:      https://resources.atlas.wooga.com/icons/icon_new.svg "New"
[ADD]:      https://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:  https://resources.atlas.wooga.com/icons/icon_improve.svg "Improve"
[CHANGE]:   https://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:      https://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:   https://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:    https://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:   https://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:      https://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:  https://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:    https://resources.atlas.wooga.com/icons/icon_webGL.svg "WebGL"
[GRADLE]:   https://resources.atlas.wooga.com/icons/icon_gradle.svg "GRADLE"
[UNITY]:    https://resources.atlas.wooga.com/icons/icon_unity.svg "Unity"
[LINUX]:    https://resources.atlas.wooga.com/icons/icon_linux.svg "Linux"
[WIN]:      https://resources.atlas.wooga.com/icons/icon_windows.svg "Windows"
[MACOS]:    https://resources.atlas.wooga.com/icons/icon_iOS.svg "macOS"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy [hunk 1]
```diff
@@ -95,19 +96,40 @@ class PaketUnityInstall extends ConventionTask implements PaketUpmPackageSpec {
...
     public final static String assemblyDefinitionFileExtension = "asmdef"
 
     /**
-     * @return the installation output directory
+     * @return The installation output directory
      */
     @OutputDirectory
     File getOutputDirectory() {
-        new File(getReferencesFile().getParentFile(), "Assets/${getPaketOutputDirectoryName()}")
+        // TODO: Update. How the heck does this work nowadays when the Unity project directory is at the root??
```
Comment (Joaquimmnetto) [inline src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy:113]: Remove this todo, or at least replace if by an comment answering it ;)

Comment (Azurelol) [reply src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy:113]: It has been refactored!!

File: src/main/groovy/wooga/gradle/paket/unity/PaketUpmPackageSpec.groovy [hunk 2]
```diff
@@ -20,6 +18,8 @@ trait PaketUpmPackageSpec implements BaseSpec {
         return paketUpmPackageManifests
     }
 
+    private final MapProperty<String, Map<String, Object>> paketUpmPackageManifests = objects.mapProperty(String, Map)
```
[the remark this answers is not in the document]
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/paket/unity/PaketUpmPackageSpec.groovy:21]: I find it much easier to parse if the first thing you see is the getter followed by the internal backing field.

[the remark this answers is not in the document]
Comment (Larusso) [reply src/main/groovy/wooga/gradle/paket/unity/PaketUpmPackageSpec.groovy:21]: I have seen many orders but never that one ;). I personally prefer the field then the getter then the setter :)

**Review by Larusso — APPROVED** [review]
It's all well and good. You only took a little bit too much liberty with renamings etc. Make the patch longer than it actually needs to.

File: src/main/groovy/wooga/gradle/paket/unity/PaketUnityPlugin.groovy [hunk 3]
```diff
@@ -56,44 +57,64 @@ class PaketUnityPlugin implements Plugin<Project> {
...
-                t.description = "Installs dependencies for Unity3d project ${referenceFile.parentFile.name} "
-                t.conventionMapping.map("paketOutputDirectoryName", { extension.getPaketOutputDirectoryName() })
+                t.description = "Installs dependencies for Unity3d project ${referencesFile.parentFile.name} "
                 t.conventionMapping.map("includeAssemblyDefinitions", { extension.getIncludeAssemblyDefinitions() })
-                t.conventionMapping.map("assemblyDefinitionFileStrategy", { extension.getAssemblyDefinitionFileStrategy() })
                 t.conventionMapping.map("preInstalledUpmPackages", { extension.getPreInstalledUpmPackages() })
                 t.frameworks = extension.getPaketDependencies().getFrameworks()
                 t.lockFile = extension.getPaketLockFile()
-                t.referencesFile = referenceFile
+                t.defaultNamespace.convention(extension.defaultNamespace)
```
Comment (Joaquimmnetto) [inline src/main/groovy/wooga/gradle/paket/unity/PaketUnityPlugin.groovy:108]: Rename this to something that makes more clear that this is about the generated UPM packages, something like `defaultUPMNamespace`

File: src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstallTask.groovy [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import org.gradle.api.tasks.OutputDirectory
+import wooga.gradle.paket.base.utils.internal.PaketLock
+import wooga.gradle.paket.base.utils.internal.PaketUPMWrapperReference
+import wooga.gradle.paket.base.utils.internal.PaketUnityReferences
+
+import java.util.function.Function
+/**
+ * A task that installs packages onto an Unity Project
+ */
+abstract class PaketUnityInstallTask extends ConventionTask {
```
Comment (Joaquimmnetto) [inline src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstallTask.groovy:16]: Mislseading name, can be confused with PaketUnityInstall.

**Review by Joaquimmnetto — APPROVED** [review]
Just naming changes, after them its alright to merge

File: src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstallTask.groovy (hunk 4 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstallTask.groovy:16]: Will make more explicit.

File: src/main/groovy/wooga/gradle/paket/unity/PaketUnityPlugin.groovy (hunk 3 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/paket/unity/PaketUnityPlugin.groovy:108]: Will do.

**Pull request merged**